### PR TITLE
chore: migrate to projen

### PR DESCRIPTION
**Pull request by iliapolo**
## Notable changes

- Tests moved to `lib/__tests__`
- [`change-control-lambda`](https://github.com/cdklabs/aws-delivlib/tree/main/change-control-lambda) moved inside `lib` instead of a separate package + use `NodeJSFunction` for bundling (avoiding the need for the `build-change-control-lambda` script)
- [`custom-resource-handlers`](https://github.com/cdklabs/aws-delivlib/tree/main/custom-resource-handlers) moved inside `lib` (but are still being built with a dedicated script).
- A bunch of ESLint fixes.

BREAKING CHANGE: `esbuild` or `docker` is required in order to bundle the change-control-lambda

-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,4 +1 @@
-
------
-
-By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
+Fixes #
```
Comment (eladb) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: revert

**Review by eladb — APPROVED** [review]
wow

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -76,13 +76,20 @@ You can use the following npm scripts to manage the pipeline:
 We have good coverage of unit tests that should be testing the bulk of the logic in delivlib. For every contribution and change,
 we expect them to be covered by unit tests, where appropriate.
 
+You can run the by executing:
```
Comment (BenChaimberg) [inline CONTRIBUTING.md:79]:
```suggestion
You can run the tests by executing:
```

Comment (iliapolo) [reply CONTRIBUTING.md:79]: 👍

**Comment by iliapolo** [comment]
@eladb It just dawned on me that the switch to `NodejsFunction` in the change-controller-lambda is a breaking change, since it requires either `esbuild` or docker to bundle. 

I'm going to mark this breaking and release a new major version. Can you acknowledge?

**Review by eladb — APPROVED** [review]
Ack

**Pull request merged**